hermes/probe/checknil/check_nil.go

```diff
@@ -0,0 +1,332 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove, GitHub: evanSpendlove
+//
+// Check_nil implements the probe operation for checking if the NIL file is
+// present and reading verifying its contents.
+
+package checknil
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strconv"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/go-cmp/cmp"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"github.com/googleinterns/step224-2020/hermes/probe"
+	"github.com/googleinterns/step224-2020/hermes/probe/metrics"
+	"google.golang.org/api/iterator"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	cpmetrics "github.com/google/cloudprober/metrics"
+	sjpb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+// nilFile stores the name, hash and contents of the nil file.
+// It is used when checking or updating the nil file.
+type nilFile struct {
+	name     string
+	hash     string
+	contents []byte
+}
+
+// validateChecksum validates that the hash matches the checksum
+// in the file name.
+// Returns:
+//	- bool: returns true if the hash matches, else false.
+func (n *nilFile) validateChecksum() bool {
+	if n.hash == "" || n.name == "" {
```

Comment:
These conditions indicate a serious bug in Hermes. I would change the signature of this function to return an `error` instead. Then split this condition and return a different error for each case.

Reply:
I've split this condition and added errors to be returned.